sharedresource/v1alpha1/types_share.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright The OpenShift authors.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Share is the Schema for the shares API
+//
+// Compatibility level 4: No compatibility is provided, the API can change at any point for any reason. These capabilities should not be used by applications needing long term support.
+// +kubebuilder:subresource:status
+// +openshift:compatibility-gen:level=4
+type Share struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+	Spec   ShareSpec   `json:"spec,omitempty" protobuf:"bytes,2,opt,name=spec"`
+	Status ShareStatus `json:"status,omitempty" protobuf:"bytes,3,opt,name=status"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ShareList contains a list of Share
+//
+// Compatibility level 4: No compatibility is provided, the API can change at any point for any reason. These capabilities should not be used by applications needing long term support.
+// +kubebuilder:subresource:status
+// +openshift:compatibility-gen:level=4
+type ShareList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+	Items           []Share `json:"items" protobuf:"bytes,2,rep,name=items"`
+}
+
+// ShareSpec defines the desired state of Share
+type ShareSpec struct {
+	// BackingResource captures the ConfigMap or Secret that is shared.
+	// +required
+	BackingResource `json:"backingResource" protobuf:"bytes,1,opt,name=backingResource"`
+
+	// Description is a user readable explanation of what the backing resource
+	// provides.
+	// +optional
+	Description string `json:"description,omitempty" protobuf:"bytes,2,opt,name=description"`
+}
+
+// ShareStatus defines the observed state of Share
+type ShareStatus struct {
+	// Conditions are the set of k8s Condition instances provided by the associated controller for Shares.
```

Comment:
I think that should be "for this Share", right?
```suggestion
	// Conditions are the set of k8s Condition instances provided by the associated controller for this Share.
```
If each Share has an associated controller, should there be a field in `ShareSpec` to specify which controller manages this share?  

If each Share can have multiple controllers, then you probably want `ShareStatus` to have a slice of values where each value has a controller name and status conditions:
```go
type ShareStatus struct {
	Controllers []ShareControllerStatus `json:"controllers,omitempty"`
}

type ShareControllerStatus struct {
	ControllerName string `json:"controllerName"`
	Conditions []metav1.Condition `json:"conditions,omitempty"`
}
```

Reply:
There's only one controller for this object - I think the godoc is misleading.

Reply:
Full disclosure, the Conditions are currently a placeholder wrt the driver/controller implementation.  No conditions are being set as of yet.

Should we perhaps remove conditions for now, and only add them if we have cause, as the implementation progresses?

Additionally, if `ShareStatus` is empty, should we even have that as a placeholder, or should we only add this as the implementation has need, etc.

WDYT @coreydaley @adambkaplan @Miciah @deads2k @otaviof ?

Reply:
Removed in next push